src/reducers/entities/roles.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {RoleTypes, UserTypes} from 'action_types';
+
+function roles(state = {}, action) {
+    const nextState = {...state};
+    switch (action.type) {
+    case RoleTypes.RECEIVED_ROLES: {
+        if (action.data) {
+            for (const role of action.data) {
+                nextState[role.name] = role;
+            }
+        }
+
+        return nextState;
```

Comment:
Minor but if you're not modifying `nextState` it's probably better to just return `state` so it's not a new object

Reply:
Fixed